### PR TITLE
feat: filters for configurator overview

### PR DESCRIPTION
**Pull request by ChristophHi**

### CODE REVIEW

File: feature-libs/product-configurator/rulebased/core/model/configurator.model.ts [hunk 1]
```diff
@@ -100,6 +100,10 @@ export namespace Configurator {
     hideBasePriceAndSelectedOptions?: boolean;
   }
 
+  export interface ConfigurationWithOverview extends Configuration {
+    overview: Overview;
+  }
```
Comment (Uli-Tiger) [inline feature-libs/product-configurator/rulebased/core/model/configurator.model.ts:105]: @ChristophHi: i created this - its used in the components - do you think this is ok?
It extends the Configuration interface, but the overview property is mandatory, which is guaranteed by having a corresponding filter in the components observable.

File: feature-libs/product-configurator/rulebased/components/overview-filter/configurator-overview-filter.component.html [hunk 2]
```diff
@@ -0,0 +1,78 @@
+<ng-container *ngIf="config$ | async as config">
+  <ng-container *ngIf="showFilterBar">
+    <cx-configurator-overview-filter-bar
+      [config]="config"
+    ></cx-configurator-overview-filter-bar>
+  </ng-container>
+  <div class="cx-overview-filter-header">
```
Comment (steinsebastian) [inline feature-libs/product-configurator/rulebased/components/overview-filter/configurator-overview-filter.component.html]: Currently the Screen-Reader only reads the text of the checkboxes (sth. like "Price Relevant Options, unticked, tick-box, ....".)

One possibility to make clear what the checkbox is about would be to connect it via aria-describedby to the header.
Therefore the header div needs to get an id and this is then referenced by aria-describedby at the input tag.

Example:
```suggestion
<div class="cx-overview-filter-header" id="cx-overview-filter-header-byOption">
  {{ 'configurator.overviewFilter.byOption' | cxTranslate }}
</div>

<div class="cx-overview-filter-option">
  <div class="form-check">
    <input
      id="cx-configurator-overview-filter-option-price"
      type="checkbox"
      class="form-check-input"
      [formControl]="priceFilter"
      (change)="onFilter(config)"
      name="config-overview-price-filter"
      aria-describedby="cx-overview-filter-header-byOption"
    />
    <label
      class="form-check-label"
      for="cx-configurator-overview-filter-option-price"
    >
      {{ 'configurator.overviewFilter.byPrice' | cxTranslate }}</label
    >
  </div>
</div>
```
Then the SR would read: "Price Relevant Options, unticked, tick-box, Filter by option....".

Another (from my point of view better) possibility would be to use a dedicated text for Screen-Readers via an aria-label.
Then the SR could read something like "Filter by Price-Relevant Options", "Filter by my selections", "Filter by Group xyz".

Example for first checkbox (new text "Filter by" for key "configurator.overviewFilter.by" has to be introduced): 
```suggestion
<input
  id="cx-configurator-overview-filter-option-price"
  type="checkbox"
  class="form-check-input"
  [formControl]="priceFilter"
  (change)="onFilter(config)"
  name="config-overview-price-filter"
  [attr.aria-label]="('configurator.overviewFilter.by'
  | cxTranslate) + ' ' + ('configurator.overviewFilter.byPrice'
  | cxTranslate)"
/>
```
This would read "Filter by Price-Relevant Options".

Example for group:
```suggestion
<input
  id="{{
    'cx-configurator-overview-filter-option-group-' + group.id
  }}"
  type="checkbox"
  class="form-check-input"
  [formControl]="groupFilters[i]"
  (change)="onFilter(config)"
  name="{{ 'config-overview-group-filter-' + group.id }}"
  [attr.aria-label]="('configurator.overviewFilter.byGroup'
  | cxTranslate) + ' ' + group.groupDescription"
/>
```

 This would read "Filter by Group xyz".

File: feature-libs/product-configurator/rulebased/components/overview-filter-button/configurator-overview-filter-button.component.ts [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+      (overview.groupFilters?.length ?? 0)
+    );
+  }
+
+  openFilterModal() {
+    this.launchDialogService.openDialogAndSubscribe(
+      LAUNCH_CALLER.CONFIGURATOR_OV_FILTER,
+      this.filerButton,
+      {}
+    );
```
Comment (steinsebastian) [inline feature-libs/product-configurator/rulebased/components/overview-filter-button/configurator-overview-filter-button.component.ts:68]: There is a problem for screen-readers when opening the modal: The focus is still at the (hidden) overview page in the background. That means after clicking the button the SR-user would still tab on the overview page and not in the filter menu. Therefore the focus needs to be set actively on an element of the modal (e.g. the close button).

**Review by steinsebastian — CHANGES_REQUESTED** [review]
Comments for A11Y for the filter menu

File: feature-libs/product-configurator/rulebased/components/overview-filter/configurator-overview-filter.component.html (hunk 2 above)
Comment (Uli-Tiger) [reply feature-libs/product-configurator/rulebased/components/overview-filter/configurator-overview-filter.component.html]: Guess i will create dedicated i18n props for the a11y texts instead of concatenating. Concatenating works with the terms we choose for englisch texts, but this might not be true for other languages.

File: feature-libs/product-configurator/rulebased/components/overview-filter-button/configurator-overview-filter-button.component.ts (hunk 3 above)
Comment (Uli-Tiger) [reply feature-libs/product-configurator/rulebased/components/overview-filter-button/configurator-overview-filter-button.component.ts:68]: how do i do that?

Comment (Uli-Tiger) [reply feature-libs/product-configurator/rulebased/components/overview-filter-button/configurator-overview-filter-button.component.ts:68]: ok found it

File: feature-libs/product-configurator/rulebased/components/overview-filter/configurator-overview-filter.component.html (hunk 2 above)
Comment (steinsebastian) [reply feature-libs/product-configurator/rulebased/components/overview-filter/configurator-overview-filter.component.html]: That's even better.

File: feature-libs/product-configurator/rulebased/core/model/configurator.model.ts (hunk 1 above)
Comment (ChristophHi) [reply feature-libs/product-configurator/rulebased/core/model/configurator.model.ts:105]: I don't like it too much. If you need the OV to be mandatory, why not have it directly as part of the method signatures where you are using ConfigurationWithOverview now?

File: feature-libs/product-configurator/rulebased/styles/_configurator-overview-sidebar.scss [hunk 4]
```diff
@@ -0,0 +1,89 @@
+%cx-configurator-overview-sidebar {
+  &:not(:empty) {
+    display: flex;
+    flex-direction: column;
+    width: 100vw;
+    margin-inline-start: 25px;
+    padding-block-start: 5px;
```
Comment (Uli-Tiger) [inline feature-libs/product-configurator/rulebased/styles/_configurator-overview-sidebar.scss]: @AlexanderKuschner: as the menu is on the left hand side and not on right hand side, the margin/padding shall be on the right hand side, as well. So it doesn't look odd. 
```suggestion
    margin-inline-end: 25px;
    padding-block-start: 5px;
```

File: feature-libs/product-configurator/rulebased/components/overview-sidebar/configurator-overview-sidebar.module.ts [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+import { ConfiguratorOverviewFilterModule } from '../overview-filter/configurator-overview-filter.module';
+import { ConfiguratorOverviewMenuModule } from '../overview-menu/configurator-overview-menu.module';
+import { ConfiguratorOverviewSidebarComponent } from './configurator-overview-sidebar.component';
+
+@NgModule({
+  imports: [
+    CommonModule,
+    I18nModule,
+    FormsModule,
+    ReactiveFormsModule,
```
Comment (Uli-Tiger) [inline feature-libs/product-configurator/rulebased/components/overview-sidebar/configurator-overview-sidebar.module.ts]: @AlexanderKuschner: do we use those 2 imports somewhere?, or a can they be removed?

File: feature-libs/product-configurator/rulebased/components/overview-sidebar/configurator-overview-sidebar.component.html [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+      {{ 'configurator.overviewSidebar.filter' | cxTranslate }}
+    </button>
+  </div>
+
+  <cx-configurator-overview-filter *ngIf="showFilter">
+  </cx-configurator-overview-filter>
+
+  <cx-configurator-overview-menu *ngIf="!showFilter">
+  </cx-configurator-overview-menu>
+</div>
```
Comment (Uli-Tiger) [inline feature-libs/product-configurator/rulebased/components/overview-sidebar/configurator-overview-sidebar.component.html]: @AlexanderKuschner: There is no component test at all for this component. I would expect a component test that toggles filter and checks what is displayed.

File: feature-libs/product-configurator/rulebased/components/overview-filter/configurator-overview-filter.module.ts [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+@NgModule({
+  imports: [
+    CommonModule,
+    I18nModule,
+    FormsModule,
+    ReactiveFormsModule,
+    ConfiguratorOverviewFilterBarModule,
+  ],
+  providers: [
+    provideDefaultConfig(<CmsConfig>{
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/overview-filter/configurator-overview-filter.module.ts]: This needs to be removed as the filter component should not be CMS managed

**Review by Uli-Tiger — CHANGES_REQUESTED** [review]
@AlexanderKuschner: can you please check my comments
@ChristophHi: your part lgtm

File: feature-libs/product-configurator/rulebased/components/overview-filter-bar/configurator-overview-filter-bar.component.html [hunk 8]
```diff
@@ -0,0 +1,58 @@
+<ng-container *ngIf="config?.overview">
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/overview-filter-bar/configurator-overview-filter-bar.component.html]: Config is always present
```suggestion
<ng-container *ngIf="config.overview">
```
@Uli-Tiger : Besides this: Why do we need this ng container at all? Don't we know that the OV is always present?

File: feature-libs/product-configurator/common/testing/common-configurator-test-utils.service.ts [hunk 9]
```diff
@@ -77,6 +77,43 @@ export class CommonConfiguratorTestUtilsService {
     expect(text ? text.trim() : '').toBe(expectedText);
   }
 
+  /**
+   * Helper function for proving whether the an elements attribute has on an value.
```
Comment (ChristophHi) [inline feature-libs/product-configurator/common/testing/common-configurator-test-utils.service.ts]: @Uli-Tiger Please fix description

File: feature-libs/product-configurator/rulebased/components/overview-filter-bar/configurator-overview-filter-bar.component.spec.ts [hunk 10]
```diff
@@ -0,0 +1,289 @@
...
+function initTestComponent() {
+  fixture = TestBed.createComponent(ConfiguratorOverviewFilterBarComponent);
+  htmlElem = fixture.nativeElement;
+  component = fixture.componentInstance;
+  component.config = ovConfig;
+  fixture.detectChanges();
+}
+
+describe('ConfiguratorOverviewFilterBarComponent', () => {
+  describe('Component Test', () => {
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/overview-filter-bar/configurator-overview-filter-bar.component.spec.ts]: @Uli-Tiger Could you rephrase that? Because it's not the 'Component test' that should do something, but a certain piece of functionality or a method.
Like 'ConfiguratorOverviewFilterBarComponent'->'in a component test environment'->'method y'->'should do x'.
Same for unit tests and other test classes

File: feature-libs/product-configurator/rulebased/components/overview-filter-bar/configurator-overview-filter-bar.module.ts [hunk 11]
```diff
@@ -0,0 +1,34 @@
...
+@NgModule({
+  imports: [
+    CommonModule,
+    I18nModule,
+    IconModule,
+    FormsModule,
+    ReactiveFormsModule,
+  ],
+  providers: [
+    provideDefaultConfig(<CmsConfig>{
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/overview-filter-bar/configurator-overview-filter-bar.module.ts]: @Uli-Tiger That's not needed as the component is not CMS managed, correct?

File: feature-libs/product-configurator/rulebased/core/model/configurator.model.ts (hunk 1 above)
Comment (ChristophHi) [reply feature-libs/product-configurator/rulebased/core/model/configurator.model.ts:105]: @Uli-Tiger But when I check it a little bit longer I understand why it helps. The suggestion that I made does not work because then we probably would also need to change the signatures of facade, actions and do changes to the effects. Let me think about it

File: feature-libs/product-configurator/rulebased/components/overview-filter-button/configurator-overview-filter-button.component.spec.ts [hunk 12]
```diff
@@ -0,0 +1,188 @@
...
+    of(ConfigurationTestData.mockRouterState)
+  );
+  asSpy(mockConfigCommonsService.getConfiguration).and.returnValue(
+    of(ovConfig)
+  );
+  asSpy(mockLaunchDialogService.openDialogAndSubscribe).and.returnValue(of());
+}
+
+describe('ConfigurationOverviewFilterButtonComponent', () => {
+  describe('Component Test', () => {
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/overview-filter-button/configurator-overview-filter-button.component.spec.ts]: See remark in previous test class, I would consider rephrasing

File: feature-libs/product-configurator/rulebased/components/overview-filter-bar/configurator-overview-filter-bar.component.ts [hunk 13]
```diff
@@ -0,0 +1,96 @@
...
+  constructor(
+    protected configuratorCommonsService: ConfiguratorCommonsService
+  ) {}
+
+  @Input() config: Configurator.ConfigurationWithOverview;
+
+  iconTypes = ICON_TYPE;
+  attributeFilterTypes = Configurator.OverviewFilter;
+
+  getGroupFilterDescription(
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/overview-filter-bar/configurator-overview-filter-bar.component.ts:32]: Add JS doc for public methods

File: feature-libs/product-configurator/rulebased/components/overview-filter-button/configurator-overview-filter-button.component.ts [hunk 14]
```diff
@@ -0,0 +1,56 @@
...
+      // filter 'strict null check safe'
+      filter(
+        (configuration) => configuration.overview != null
+      ) as OperatorFunction<
+        Configurator.Configuration,
+        Configurator.ConfigurationWithOverview
+      >
+    );
+
+  getNumFilters(overview: Configurator.Overview): number {
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/overview-filter-button/configurator-overview-filter-button.component.ts:52]: Add JS Doc for public methods

File: feature-libs/product-configurator/rulebased/components/overview-filter/configurator-overview-filter.component.ts [hunk 15]
```diff
@@ -0,0 +1,128 @@
...
+        Configurator.ConfigurationWithOverview
+      >,
+      tap((configuration: Configurator.ConfigurationWithOverview) => {
+        this.extractAttrFilterState(configuration);
+        this.extractGroupFilterState(configuration);
+      })
+    );
+
+  onFilter(config: Configurator.ConfigurationWithOverview) {
+    let inputConfig = this.createInputConfig(
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/overview-filter/configurator-overview-filter.component.ts]: @Uli-Tiger Why not 'const' here?

File: feature-libs/product-configurator/rulebased/components/overview-filter/configurator-overview-filter.component.ts [hunk 16]
```diff
@@ -0,0 +1,128 @@
...
+      isSelected =
+        configuration.overview.attributeFilters.indexOf(
+          Configurator.OverviewFilter.USER_INPUT
+        ) >= 0;
+      this.mySelectionsFilter.setValue(isSelected);
+    }
+  }
+
+  protected collectGroupFilters(overview: Configurator.Overview): string[] {
+    let filters: string[] = [];
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/overview-filter/configurator-overview-filter.component.ts]: @Uli-Tiger 
```suggestion
    const filters: string[] = [];
```

File: feature-libs/product-configurator/rulebased/components/overview-filter/configurator-overview-filter.component.ts [hunk 17]
```diff
@@ -0,0 +1,128 @@
...
+      if (groupFilter.value && overview?.possibleGroups) {
+        filters.push(overview.possibleGroups[idx].id);
+      }
+      idx++;
+    });
+    return filters;
+  }
+
+  protected collectAttrFilters(): Configurator.OverviewFilter[] {
+    let filters: Configurator.OverviewFilter[] = [];
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/overview-filter/configurator-overview-filter.component.ts]:
```suggestion
    const filters: Configurator.OverviewFilter[] = [];
```

**Review by ChristophHi — COMMENTED** [review]
@Uli-Tiger Styling looks odd on small devices
Screenshot 2022-12-02 at 12 01 27

**Review by ChristophHi — COMMENTED** [review]
@Larisa-Staroverova Could you check why the sidebar is no longer sticky?

File: feature-libs/product-configurator/rulebased/components/overview-filter-bar/configurator-overview-filter-bar.component.ts [hunk 18]
```diff
@@ -0,0 +1,96 @@
...
+        ?.groupDescription ?? ''
+    );
+  }
+
+  onAttrFilterRemove(
+    config: Configurator.ConfigurationWithOverview,
+    attrToRemove: string
+  ) {
+    let attrFilters = config.overview.attributeFilters ?? [];
+    let groupFilters = config.overview.groupFilters ?? [];
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/overview-filter-bar/configurator-overview-filter-bar.component.ts]: @Uli-Tiger 
```suggestion
    const groupFilters = config.overview.groupFilters ?? [];
```

File: feature-libs/product-configurator/rulebased/components/overview-filter-bar/configurator-overview-filter-bar.component.ts [hunk 19]
```diff
@@ -0,0 +1,96 @@
...
+  @Input() config: Configurator.ConfigurationWithOverview;
+
+  iconTypes = ICON_TYPE;
+  attributeFilterTypes = Configurator.OverviewFilter;
+
+  getGroupFilterDescription(
+    overview: Configurator.Overview,
+    groupId: string
+  ): string {
+    console.log(overview);
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/overview-filter-bar/configurator-overview-filter-bar.component.ts]: @Uli-Tiger Please remove log statement

File: feature-libs/product-configurator/rulebased/components/overview-filter-bar/configurator-overview-filter-bar.component.ts [hunk 20]
```diff
@@ -0,0 +1,96 @@
...
+      overview.possibleGroups?.find((group) => group.id === groupId)
+        ?.groupDescription ?? ''
+    );
+  }
+
+  onAttrFilterRemove(
+    config: Configurator.ConfigurationWithOverview,
+    attrToRemove: string
+  ) {
+    let attrFilters = config.overview.attributeFilters ?? [];
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/overview-filter-bar/configurator-overview-filter-bar.component.ts]: @Uli-Tiger The fallback is not covered by tests

File: feature-libs/product-configurator/rulebased/components/overview-filter-bar/configurator-overview-filter-bar.component.ts [hunk 21]
```diff
@@ -0,0 +1,96 @@
...
+    this.configuratorCommonsService.updateConfigurationOverview(
+      this.createInputConfig(config, attrFilters, groupFilters)
+    );
+  }
+
+  onGroupFilterRemove(
+    config: Configurator.ConfigurationWithOverview,
+    groupIdToRemove: string
+  ) {
+    let attrFilters = config.overview.attributeFilters ?? [];
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/overview-filter-bar/configurator-overview-filter-bar.component.ts]: @Uli-Tiger 
```suggestion
    const attrFilters = config.overview.attributeFilters ?? [];
```

File: feature-libs/product-configurator/rulebased/components/overview-filter-bar/configurator-overview-filter-bar.component.ts [hunk 22]
```diff
@@ -0,0 +1,96 @@
...
+    let groupFilters = config.overview.groupFilters ?? [];
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/overview-filter-bar/configurator-overview-filter-bar.component.ts]: @Uli-Tiger Fallback not covered by tests

File: feature-libs/product-configurator/rulebased/components/overview-filter-bar/configurator-overview-filter-bar.component.ts [hunk 23]
```diff
@@ -0,0 +1,96 @@
...
+      (groupId) => groupIdToRemove !== groupId
+    );
+
+    this.configuratorCommonsService.updateConfigurationOverview(
+      this.createInputConfig(config, attrFilters, groupFilters)
+    );
+  }
+
+  isShowRemoveAll(overview: Configurator.Overview): boolean {
+    let numFilters =
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/overview-filter-bar/configurator-overview-filter-bar.component.ts]: @Uli-Tiger 
```suggestion
    const numFilters =
```

**Review by ChristophHi — COMMENTED** [review]
@Uli-Tiger , @AlexanderKuschner : We need to expose the new components here: /rulebased/components/index.ts, otherwise customers cannot extend.
Will do that, I also forgot it for the component I introduced

File: feature-libs/product-configurator/rulebased/components/overview-sidebar/configurator-overview-sidebar.component.html [hunk 24]
```diff
@@ -0,0 +1,16 @@
+<div class="cx-sticky-menu">
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/overview-sidebar/configurator-overview-sidebar.component.html]: @AlexanderKuschner This component is missing a ghost animation. The navigation is visible already when data is still loading

File: feature-libs/product-configurator/common/assets/translations/en/configurator-common.ts [hunk 25]
```diff
@@ -227,6 +239,16 @@ export const configurator = {
...
       addToCartPrices:
         'Base Price {{basePrice}}, Selected Options {{selectedOptions}}, Total Price {{totalPrice}}',
+      filterOverview: 'Open configuration overview filter menu',
+      filterOverviewWithCount:
+        'Open the configuration overview filter menu - {{numAppliedFilters}} active filters',
+      closeFilterMenu:
+        'Close the configuration overview filter menu and apply the selected filters',
+      filterOverviewByPrice: 'Filter configuration overview by your selections',
+      filterOverviewByMySelections:
+        'Filter configuration overview by price-relevant options',
```
Comment (steinsebastian) [inline feature-libs/product-configurator/common/assets/translations/en/configurator-common.ts:248]: @Uli-Tiger The text does not fit the key (both for your selections and price-relevant options. It just has to be swapped.

File: feature-libs/product-configurator/rulebased/components/overview-filter-bar/configurator-overview-filter-bar.component.html (hunk 8 above)
Comment (Uli-Tiger) [reply feature-libs/product-configurator/rulebased/components/overview-filter-bar/configurator-overview-filter-bar.component.html]: first i thought you are right, but if i remove it UI crashes on navigation to OV with
```
ERROR TypeError: ctx.config is undefined
    ConfiguratorOverviewFilterBarComponent_Template configurator-overview-filter-bar.component.html:2
```
maybe we can have a look to it on monday

File: feature-libs/product-configurator/rulebased/components/overview-sidebar/configurator-overview-sidebar.component.html [hunk 26]
```diff
@@ -0,0 +1,16 @@
...
+  <div class="cx-menu-bar">
```
Comment (steinsebastian) [inline feature-libs/product-configurator/rulebased/components/overview-sidebar/configurator-overview-sidebar.component.html]: @AlexanderKuschner The enclosing div needs to get the role tablist
```suggestion
  <div class="cx-menu-bar" role="tablist">
```

File: feature-libs/product-configurator/rulebased/components/overview-sidebar/configurator-overview-sidebar.component.html [hunk 27]
```diff
@@ -0,0 +1,16 @@
...
+    <button role="tab" [class.active]="!showFilter" (click)="onMenu()">
```
Comment (steinsebastian) [inline feature-libs/product-configurator/rulebased/components/overview-sidebar/configurator-overview-sidebar.component.html]: @AlexanderKuschner The active tab has to be marked for screen-readers with aria-selected.
```suggestion
    <button role="tab" [attr.aria-selected]="!showFilter" [class.active]="!showFilter" (click)="onMenu()">
```

File: feature-libs/product-configurator/rulebased/components/overview-sidebar/configurator-overview-sidebar.component.html [hunk 28]
```diff
@@ -0,0 +1,16 @@
...
+      {{ 'configurator.overviewSidebar.menu' | cxTranslate }}
+    </button>
+    <button role="tab" [class.active]="showFilter" (click)="onFilter()">
```
Comment (steinsebastian) [inline feature-libs/product-configurator/rulebased/components/overview-sidebar/configurator-overview-sidebar.component.html]: @AlexanderKuschner The active tab has to be marked for screen-readers with aria-selected.
```suggestion
    <button role="tab" [attr.aria-selected]="showFilter" [class.active]="showFilter" (click)="onFilter()">
```

**Comment by Uli-Tiger** [comment]
current components reading OV-Data (3 components)
```
menu

filter
  |-> filter-bar

filter-button 
  |-> filter-bar
```

could be simplified to (2 components)
```
sidebar
  | -> menu
  | -> filter
           | -> filter-bar
filter-button
  | -> filter-bar
  | -> filter-dialog
           | -> filter 
                   |-> filter-bar
```

File: feature-libs/product-configurator/rulebased/components/overview-filter-bar/configurator-overview-filter-bar.component.html (hunk 8 above)
Comment (Uli-Tiger) [reply feature-libs/product-configurator/rulebased/components/overview-filter-bar/configurator-overview-filter-bar.component.html]: after the refactoring i could remove this completly

File: feature-libs/product-configurator/rulebased/components/overview-filter-bar/configurator-overview-filter-bar.component.ts [hunk 29]
```diff
@@ -0,0 +1,128 @@
...
+
+    this.configuratorCommonsService.updateConfigurationOverview(
+      this.createInputConfig(config, attrFilters, groupFilters)
+    );
+  }
+
+  /**
+   * check whether the button to remove all filters should be shown
+   *
+   * @param {Configurator.ConfigurationWithOverview} config - current configuration with overview data
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/overview-filter-bar/configurator-overview-filter-bar.component.ts]: @Uli-Tiger Please correct, does not match signature

File: feature-libs/product-configurator/rulebased/components/overview-filter-button/configurator-overview-filter-button.component.ts [hunk 30]
```diff
@@ -0,0 +1,66 @@
...
+  templateUrl: './configurator-overview-filter-button.component.html',
+})
+export class ConfiguratorOverviewFilterButtonComponent {
+  constructor(
+    protected launchDialogService: LaunchDialogService,
+    protected configuratorCommonsService: ConfiguratorCommonsService,
+    protected configRouterExtractorService: ConfiguratorRouterExtractorService
+  ) {}
+
+  @ViewChild('filterButton') filerButton: ElementRef;
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/overview-filter-button/configurator-overview-filter-button.component.ts]: @Uli-Tiger Please rename
```suggestion
  @ViewChild('filterButton') filterButton: ElementRef;
```

File: feature-libs/product-configurator/rulebased/components/overview-filter-button/configurator-overview-filter-button.component.ts [hunk 31]
```diff
@@ -0,0 +1,66 @@
...
+      ) as OperatorFunction<
+        Configurator.Configuration,
+        Configurator.ConfigurationWithOverview
+      >
+    );
+
+  /**
+   * get the number of filters currently applied to the overview page
+   *
+   * @param {Configurator.ConfigurationWithOverview} config - current configuration with overview data
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/overview-filter-button/configurator-overview-filter-button.component.ts]: @Uli-Tiger Please change, does not match signature

File: feature-libs/product-configurator/rulebased/root/variant/variant-configurator-overview-layout.module.ts [hunk 32]
```diff
@@ -21,20 +21,7 @@ import { VariantConfiguratorPageLayoutHandler } from './variant-configurator-pag
       layoutSlots: {
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/root/variant/variant-configurator-overview-layout.module.ts:21]: @Larisa-Staroverova I know that this is not directly related to this file, but: We need to ensure that the OV page looks nice also if the new sidebar slot is not available (older commerce release!). So the solution we have here and in feature-libs/product-configurator/rulebased/styles/pages/_configurator-variant-overview-page.scss is not sufficient

File: feature-libs/product-configurator/rulebased/components/overview-filter-button/configurator-overview-filter-button.component.html [hunk 33]
```diff
@@ -0,0 +1,24 @@
+<ng-container *ngIf="config$ | async as config">
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/overview-filter-button/configurator-overview-filter-button.component.html]: @Uli-Tiger , @Larisa-Staroverova I think we need a loading animation here as well, do you agree? (Missing animation is not very visible though, as the OV form loading animation is present)

File: feature-libs/product-configurator/rulebased/components/overview-sidebar/configurator-overview-sidebar.component.ts [hunk 34]
```diff
@@ -0,0 +1,47 @@
...
+      // filter 'strict null check safe'
+      filter(
+        (configuration) => configuration.overview != null
+      ) as OperatorFunction<
+        Configurator.Configuration,
+        Configurator.ConfigurationWithOverview
+      >
+    );
+
+  onFilter() {
```
Comment (ChristophHi) [inline feature-libs/product-configurator/rulebased/components/overview-sidebar/configurator-overview-sidebar.component.ts:51]: @AlexanderKuschner Please add JS doc for public methods

File: feature-libs/product-configurator/rulebased/components/overview-filter-button/configurator-overview-filter-button.component.html (hunk 33 above)
Comment (Uli-Tiger) [reply feature-libs/product-configurator/rulebased/components/overview-filter-button/configurator-overview-filter-button.component.html]: added a ghost animation

File: feature-libs/product-configurator/rulebased/styles/pages/_configurator-variant-overview-page.scss [hunk 35]
```diff
@@ -14,6 +14,10 @@ $splitConfiguratorOvPage: true !default;
       @include media-breakpoint-up(lg) {
         cx-page-slot.VariantConfigOverviewSidebar {
           max-inline-size: 30%;
+          position: -webkit-sticky;
+          position: sticky;
+          top: 0;
+          marging-block-end: 50px;
```
Comment (Uli-Tiger) [inline feature-libs/product-configurator/rulebased/styles/pages/_configurator-variant-overview-page.scss]: @Larisa-Staroverova: there is a typo - should be `margin-block-end: 75px;`
and i think we need 75px - roughly the height of the add-to-cart button bar

**Review by Larisa-Staroverova — DISMISSED** [review]

**Review by Uli-Tiger — DISMISSED** [review]

**Review by Larisa-Staroverova — DISMISSED** [review]

**Review by Uli-Tiger — DISMISSED** [review]

**Review by steinsebastian — DISMISSED** [review]

**Review by Uli-Tiger — CHANGES_REQUESTED** [review]
now i do not see the sidebar anymore....

**Review by Larisa-Staroverova — APPROVED** [review]

**Review by Uli-Tiger — APPROVED** [review]
